src/components/SearchBar/SearchBar.vue

```diff
@@ -0,0 +1,154 @@
+<template>
+  <div ref="searchBar">
+    <text-input
+      id="searchBar"
+      v-model="searchTerm"
+      class="w-max bg-white-300 h-12"
+      grey
+    >
+      <template #iconLeft>
+        <search class="w-4 h-6" />
+      </template>
+      <template #iconRight>
+        <button
+          :role="disabled ? 'button' : null"
+          aria-label="Close"
+          :disabled="disabled"
+          data-testid="clearSearchButton"
+          @click="clearSearch"
+        >
+          <close class="w-4 h-6" />
+        </button>
+      </template>
+    </text-input>
+    <div
+      v-if="!outsideClick"
+      class="bg-white shadow overflow-y-auto max-h-56"
+    >
+      <div
+        v-if="searchTerm"
+        class="text-center py-4"
+      >
+        <template v-if="searching">
+          Loading, please wait...
+        </template>
+        <template v-else-if="searchResults.length">
+          View all {{ searchResults.length }} results...
+        </template>
+        <template v-else>
+          No results found
+        </template>
+      </div>
+      <Table
+        v-if="!searching && searchResults.length"
+        class="min-w-full divide-y divide-gray-200"
+        space="sm"
+      >
+        <TableBody>
+          <TableRow
+            v-for="result in searchResults"
+            :key="result"
+            class="hover:shadow rounded-md cursor-pointer"
+          >
+            <div
+              v-for="key in Object.keys(result)"
+              :key="key"
+              class="whitespace-nowrap"
+            >
+              <img
+                v-if="key === 'img'"
+                :src="result[key]"
+                class="w-5 h-5"
+              >
+              <template v-else>
+                {{ result[key] }}
+              </template>
+            </div>
+            <div class="text-right text-xl">
+              >
+            </div>
+          </TableRow>
+        </TableBody>
+      </Table>
+    </div>
+  </div>
+</template>
+
+<script>
+import textInput from '../TextInput/TextInput';
```

Comment:
This is a nitpick but can we use consistent casing in our imports?

Reply:
And we can use absolute paths for more resilience in the future (in case we move files around)
```
import { Search, Close, TextInput, Table, TableBody, TableRow } from '@/components'
```

(though it does look like the Table exports are missing from `src/components/index.js` so could you add those while you're in there?)

Reply:
@mercedesb agreed, this looks cleaner and safer. However, we experienced a strange issue when doing this, which you can further study from this issue: https://github.com/storybookjs/storybook/issues/9516.

We only got it working if in `index.js`, the export for searchBar is at the bottom. We cannot import components that appear further below in the list.

Reply:
What a weird bug!